src/controllers/TalksController.php

```diff
@@ -140,8 +142,229 @@ public function postAction($request, $db) {
+                throw new Exception(implode(". ", $errors), 400);
+            }
+
+
+            $talk['start_date'] = (new \DateTime($request->getParameter("start_date")))->format('U');
+
+            $talk['duration'] = filter_var(
+                $request->getParameter('duration'),
+                FILTER_SANITIZE_NUMBER_INT
+            );
+            if (empty($talk['duration'])) {
+                $talk['duration'] = 60;
+            }
+
+            $talk['language'] = filter_var($request->getParameter('language'), FILTER_SANITIZE_STRING);
+            if (empty($talk['language'])) {
+                $talk['language'] = 'English - UK';
+            }
+            // TODO: @heiglandreas Check whether the language actually exists in the database
+            // When the language doesn't exist, the talk will not be found
+
+            $talk['slides_link'] = filter_var($request->getParameter('slides_link'), FILTER_SANITIZE_URL);
+
+            $incoming_speakers_list = $request->getParameter('speakers');
+            if(is_array($incoming_speakers_list)) {
+                $talk['speakers'] = array_map(function($speaker){
+                    $speaker = filter_var($speaker, FILTER_SANITIZE_STRING);
+                    $speaker = trim($speaker);
+                    return $speaker;
+                }, $incoming_speakers_list);
+            }
+
+            $talk_mapper = new TalkMapper($db, $request);
+            $new_id = $talk_mapper->save($talk);
+
+            $uri = $request->base . '/' . $request->version . '/talks/' . $new_id;
+            header("Location: " . $uri, true, 201);
+            exit;
+        //    throw new Exception("method not yet supported - sorry");
+        }
+    }
+
+    /**
+     * Edit a talk
+     *
+     * This action expects the following parameters:
+     *
+     * * int talk_id
+     * * int event_id
+     * * string title
+     * * string url_friendly_talk_title
+     * * string description
+     * * string language (has to be a value from the langauges-table
+     * * string slides_link (has to be a URL)
+     * * string type (has to be a value from the categories-table
+     * * int duration
+     * * string date (will be parsed as date)
+     * * string[] speakers (speaker names as the fullnames in the user-table)
+     *
+     * Users with admin-rights on the talk (the admins as well as the event-hosts)
+     * can edit all fields, verified speakers will only be able to edit title,
+     * description, language and slides-link.
+     *
+     * @param $request
+     * @param $db
+     *
+     * @throws Exception
+     */
+    public function putAction($request, $db)
+    {
+        $errors = array();
+        if(!isset($request->user_id)) {
+            throw new Exception("You must be logged in to edit data", 400);
+        }
+        $talk_id = $this->getItemId($request);
+
+        if (isset($request->url_elements[4])) {
+            // Not editing the talk but something "below"
```

Comment:
You should be able to craft the `path` in `2.1.json` so that we don't need this test at all. i.e. let's ensure that the router doesn't match sub-resources to this action method.